src/type_operators.rs

```diff
@@ -281,6 +282,9 @@ fn pow_test() {
 pub trait Cmp<Rhs = Self> {
     /// The result of the comparison. It should only ever be one of `Greater`, `Less`, or `Equal`.
     type Output;
+
+    #[doc(hidden)]
+    fn compare<IM: InternalMarker>(&self, &Rhs) -> Self::Output;
```

Comment:
Since `InternalMarker` is only implemented by one crate-local type, `Internal`, it is not possible for end-users to call this method.

Reply:
Neat trick!

I'm not sure it's needed here, though. At least, I see no harm in exposing this method.

Reply:
I used this trick just to avoid committing to any permanent decisions about whether this method (and others) should take its parameters by value or reference.

For instance, if helping the compiler optimize these method calls away is a priority, then by-value might be preferable. Alternatively, if making typenum with `dyn Unsigned` values is a goal, then by-ref is generally preferable. I don't know which set of tradeoffs are preferable, so `InternalMarker` leaves the decision open.

Reply:
Fair enough. I don't think it matters much, though. I don't expect these functions to be called much, as this library is really about the type operations -- the types used to not even be instantiable.

But I'm happy with leaving this hidden until / unless someone asks for it to be exposed.